rclrs/src/node/builder.rs

```diff
@@ -192,4 +271,47 @@ impl NodeBuilder {
             subscriptions: std::vec![],
         })
     }
+
+    /// Creates node options.
+    /// options are overrided by field values of builder.    
+    fn create_node_options(&self) -> Result<rcl_node_options_t, RclrsError> {
+        // SAFETY: No preconditions for this function.
+        let mut node_options = unsafe { rcl_node_get_default_options() };
+
+        let cstring_args = self
+            .arguments
+            .iter()
+            .map(|s| CString::new(s.as_str()).unwrap())
+            .collect::<Vec<_>>();
+        let cstring_arg_ptrs = cstring_args.iter().map(|s| s.as_ptr()).collect::<Vec<_>>();
+        // SAFETY: Getting a zero-initialized value is always safe.
+        unsafe {
+            // SAFETY: This function does not store the ephemeral cstring_args_ptrs
+            // pointers. We are passing in a zero-initialized arguments struct as expected.
+            rcl_parse_arguments(
+                cstring_arg_ptrs.len() as i32,
+                cstring_arg_ptrs.as_ptr(),
+                rcutils_get_default_allocator(),
+                &mut node_options.arguments,
+            )
+        }
+        .ok()?;
+
+        node_options.use_global_arguments = self.use_global_arguments;
+        node_options.enable_rosout = self.enable_rosout;
+        // SAFETY: No preconditions for this function.
+        node_options.allocator = unsafe { rcutils_get_default_allocator() };
+
+        Ok(node_options)
+    }
+}
+
+impl Drop for rcl_node_options_t {
+    fn drop(&mut self) {
+        // SAFETY: Do not finish this struct except here.
+        unsafe {
+            // This also finalizes the `rcl_arguments_t` contained in `rcl_node_options_t`,
```

Comment:
```suggestion
            // This also finalizes the `rcl_arguments_t` contained in `rcl_node_options_t`.
```

Super nitpicky, I know :)